fetch/cosmwasm_contract/src/contract.rs

```diff
@@ -30,7 +32,29 @@ pub fn instantiate(
     info: MessageInfo,
     msg: InstantiateMsg,
 ) -> StdResult<Response> {
-    let env_message_sender = info.sender;
+    let supply = amount_from_funds(&info.funds, msg.get_denom()).unwrap_or(Uint128::zero());
+    initialise_contract_state(
+        deps.storage,
+        &env,
+        &info.sender,
+        supply,
+        DEFAULT_RELAY_EON,
+        DEFAULT_FEES_ACCRUED,
+        &msg,
+    )?;
+
+    Ok(Response::default())
+}
+
+pub fn initialise_contract_state(
```

Comment:
Maybe it is ok because this migration will be one-time thing, but this means that during migration we will have to set all parameters to contract manually and it will rewrite the original contract State.